### PR TITLE
chore: Auto-update replicas for conference day

### LINKED ISSUE #3582: [for ECS environment] scale out replicas in conferences days automatically
We warm up replicas for `dreamkast`, `dreamkast-ui`, etc... containers before every conference in EKS environment.
We want to realize the automatically warm-up operation in ECS environment.

### Tasks

* execute cron-job (running on GHA, ECS Scheduled Task, or others) as the following
    * get event schedule from https://event.cloudnativedays.jp/api/v1/events
    * if today is conference day
        * add line `replicas=10,` in the following file
            * https://github.com/cloudnativedaysjp/dreamkast-infra/blob/a0c7d7a9d8d9aa260353fe3c9dc439f8543f0f75/ecspresso/stg/dreamkast/service-def.jsonnet#L4-L10
        * push the update to `main` branch
            * then, deployed it automatically (by [this GHA](https://github.com/cloudnativedaysjp/dreamkast-infra/blob/main/.github/workflows/ecspresso.yml))
    * else
        * remove line `replicas=10,`
        * push the update to `main` branch

### References

In dreamkast EKS environment, we developed https://github.com/cloudnativedaysjp/dreamkast-external-scaler to scale out replicas in conference days automatically.

We need to develop instead of it.

### PR DESCRIPTION
**Pull request by mayocream**
# Description

Fixes #3582 (issue)

## Type of change

- [ ] Document update or simple typo fix
- [ ] Maintenance/update components
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] I have checked backward/forward compatibility that may cause regarding this change.

### CODE REVIEW

File: .github/workflows/autoscale.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            const today = new Date().toISOString().split('T')[0]
+            const isConferenceDay = events.some(event =>
+              event.conferenceDays.some(day => day.date === today)
+            )
+            return isConferenceDay
+
+      - name: Update service-def.jsonnet
+        run: |
+          is_conference_day="${{ steps.is_conference_day.outputs.result }}"
+          jsonnet_file="ecspresso/stg/dreamkast/service-def.jsonnet"
```
Comment (ShotaKitazawa) [inline .github/workflows/autoscale.yml:32]: 1. not only `dreamkast` but also `dreamkast-ui`, `dreamkast-weaver` should be updated.
2. A new directory `ecspresso/prod/dreamkast/service-def.jsonnet` will be created in the future. And, this action should update this file.
    * please set "ENVIRONMENT=prod" as environment variable and use this environment variable here.

File: .github/workflows/autoscale.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+          if [ "$is_conference_day" == "true" ]; then
+            sed -i '/replicas=[0-9]*/d' "$jsonnet_file"
```
Comment (ShotaKitazawa) [inline .github/workflows/autoscale.yml:35]: (nits) It's good, but `sed` command is not a bit readable I think.
How about describing `replicas` at another file and Action updating this file?

* https://github.com/cloudnativedaysjp/dreamkast-infra/commit/21c17d4d4f3bcdf50a647f41db0824c9035c9f4a

**Comment by ShotaKitazawa** [comment]
Thank you for your working!
Not to enough information in my past instructions, so I commented in some lines.

(私は英語が苦手なので、誤解のある表現をしていたら遠慮なく指摘してください🙇)

**Comment by mayocream** [comment]
> Thank you for your working! Not to enough information in my past instructions, so I commented in some lines.
> 
> (私は英語が苦手なので、誤解のある表現をしていたら遠慮なく指摘してください🙇)

ご指摘ありがとうございます！！！日本語で大丈夫ので今後日本語でIssue作ってもOKです！

File: .github/workflows/autoscale.yml (hunk 2 above)
Comment (mayocream) [reply .github/workflows/autoscale.yml:35]: Thanks for the suggestion! It shoud be more easy to mantain.

File: .github/workflows/autoscale.yml (hunk 1 above)
Comment (mayocream) [reply .github/workflows/autoscale.yml]: 1. updated!
2. follow-upのPRを作りました; `prod` folder 実際に作った後にmergeしましょう、その前にstgでテストすることが出来ます　https://github.com/cloudnativedaysjp/dreamkast-infra/pull/3616

**Review by ShotaKitazawa — APPROVED** [review]
LGTM!

**Pull request merged**